component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java

```diff
@@ -421,51 +724,33 @@ private CarbonMessage generateCarbonMessage(List<Header> headers, HttpPayloadDat
          */
         // Set protocol type http or https
         cMessage.setProperty(Constants.PROTOCOL, httpURLProperties.get(HttpConstants.SCHEME));
-            // Set uri
+        // Set uri
         cMessage.setProperty(Constants.TO, httpURLProperties.get(HttpConstants.TO));
-            // set Host
+        // set Host
         cMessage.setProperty(Constants.HOST, httpURLProperties.get(HttpConstants.HOST));
-            //set port
+        //set port
         cMessage.setProperty(Constants.PORT, Integer.valueOf(httpURLProperties.get(HttpConstants.PORT)));
-            // Set method
+        // Set method
         cMessage.setProperty(HttpConstants.HTTP_METHOD, httpMethod);
 
         /*
          *set request headers.
          */
-            // Set user given Headers
+        // Set user given Headers
         if (headers != null) {
-            cMessage.setHeaders(headers);
+            HttpHeaders httpHeaders = cMessage.getHeaders();
+            for (Header header : headers) {
+                httpHeaders.set(header.getName(), header.getValue());
+            }
         }
-            // Set content type if content type s not included in headers
+        // Set content type if content type s not included in headers
         if (contentType.contains(mapType)) {
             cMessage.setHeader(HttpConstants.HTTP_CONTENT_TYPE, contentType);
         }
-            //set content length header
-        if (cMessage.getHeaders().get(HttpConstants.CONTENT_LENGTH_HEADER) == null) {
-            try {
-                cMessage.setHeader(HttpConstants.CONTENT_LENGTH_HEADER, String.valueOf(payloadString.getBytes
-                        (HttpConstants.DEFAULT_ENCODING).length));
-            } catch (UnsupportedEncodingException e) {
-                log.error("Unsupported content encoding.");
-            }
-        }
-            //set method-type header
-        cMessage.setHeader(HttpConstants.HTTP_METHOD, httpMethod);
 
-        /*
-         * set request payload.
-         */
-                // Set message body
-        if (payload != null) {
-            payload.setOutputStream(cMessage.getOutputStream());
-            cMessage.setMessageDataSource(payload);
-            cMessage.setAlreadyRead(true);
-        }
-                //Handel Empty Messages
-        if (cMessage.isEmpty() && cMessage.getMessageDataSource() == null) {
-            cMessage.setEndOfMsgAdded(true);
-        }
+        //set method-type header
+        cMessage.setHeader(HttpConstants.HTTP_METHOD, httpMethod);
+        //cMessage.setEndOfMsgAdded(true);
```

Comment:
Any reason for this being commented out?

Reply:
Fixed in dd6916be405bc6fc81078deffdfe7b687a1bf283